pkg/cloud/azure/azure_storage.go

```diff
@@ -169,27 +182,48 @@ func makeAzureStorage(
 		return nil, errors.Wrap(err, "azure: account name is not valid")
 	}
 
-	//TODO(benbardin): Implicit auth.
 	var azClient *service.Client
-	if conf.ClientID != "" {
-		credential, err := azidentity.NewClientSecretCredential(conf.TenantID, conf.ClientID, conf.ClientSecret, nil)
-		if err != nil {
-			return nil, errors.Wrap(err, "azure client secret credential")
-		}
+	switch conf.Auth {
+	case "", cloud.AuthParamSpecified:
```

Comment:
Given my suggestion above, this would read easier:
```
switch conf.AzureAuth
case Implicit:

case LegacyKey:

case RoleBased:
```

Reply:
done